perfact/zodbsync/object_types.py

```diff
@@ -402,26 +402,38 @@ def create(obj, data, obj_id):
     @staticmethod
     def read(obj):
         # late additions may not yet be everywhere in the Data.fs
-        return {
+        obj_dict = {
             'autocommit': getattr(obj, 'autocommit', False),
             'readonlymode': getattr(obj, 'readonlymode', False),
             'connection_string': obj.connection_string,
             'encoding': obj.encoding,
             'tilevel': obj.tilevel,
             'zdatetime': obj.zdatetime,
         }
+        # Place additional parameters into the object dict only if
+        # they're set
+        if getattr(obj, 'use_tpc', False):
```

Comment:
Won't work this way.

If I am using a new system and a connector currently has this flag set and a change I want to transmit turns it off, this will be removed from the recording and playing it back will not remove the flag but leave it as it is.
We need to use `hasattr` here.